parser.y

```diff
@@ -9634,6 +9634,10 @@ NumericType:
 		// TODO: check flen 0
 		x := types.NewFieldType($1.(byte))
 		x.Flen = $2.(int)
+		if $2.(int) != types.UnspecifiedLength {
+			yylex.AppendError(yylex.Errorf("Integer display width is deprecated and will be removed in a future release."))
```

Comment:
Could we use `ErrWarnDeprecatedIntegerDisplayWidth ` directly?

Reply:
it can't use a `terror` class directly in scanner here,  `ErrWarnDeprecatedIntegerDisplayWidth` is used in TiDB to append warnings in statement context to replace the raw warning, it contains the error code 1681, error message. 
the raw message is like: "You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use near ''%s" Integer display width is deprecated and will be removed in a future release" and the fault error code is 1064